appendices/VK_EXT_present_timing.adoc

```diff
@@ -0,0 +1,249 @@
+// Copyright (c) 2017-2023 Khronos Group.
+//
+// SPDX-License-Identifier: CC-BY-4.0
+
+include::{generated}/meta/{refprefix}VK_EXT_present_timing.adoc[]
+
+=== Other Extension Metadata
+
+*Last Modified Date*::
+    2023-03-07
+*IP Status*::
+    No known IP claims.
+*Contributors*::
+  - Ian Elliott, Google
+  - James Jones, NVIDIA
+  - Jeff Juliano, NVIDIA
+  - Daniel Rakos, AMD
+  - Daniel Stone, Collabora
+  - Daniel Vetter, Intel
+  - Aric Cyr, AMD
+  - Faith Ekstrand, Intel
+  - Nicolai Hähnle, AMD
+  - Alon Or-Bach, Samsung
+  - Niklas Smedberg, Unity Technologies
+  - Tobias Hector, AMD
+  - Lionel Duc, NVIDIA
+  - Lina Versace, Google
+  - Sebastian Wick, Red Hat
+  - Jakob Bornecrantz, Collabora
+
+=== Description
+
+This device extension allows an application that uses the
+`apiext:VK_KHR_swapchain` extension to obtain information about the
+presentation engine's display, to obtain timing information about each
+present, and to schedule a present to happen no earlier than a desired time.
+An application can use this to minimize various visual anomalies (e.g.
+stuttering).
+
+Traditional game and real-time animation applications need to correctly
+position their geometry for when the presentable image will be presented to
+the user.
+To accomplish this, applications need various timing information about the
+presentation engine's display.
+They need to know when presentable images were actually presented, and when
+they could have been presented.
+Applications also need to tell the presentation engine to display an image
+no sooner than a given time.
+This allows the application to avoid stuttering, so the animation looks
+smooth to the user.
+
+include::{generated}/interfaces/VK_EXT_present_timing.adoc[]
+
+=== Issues
+
+1) How does the application determine refresh duration, quanta for change,
+whether FRR vs. VRR, etc.
+
```

Comment:
@nvlduc @cubanismo and @versalinyaa FYI ... Android has 2 other types of displays, at least one of which we'll want to add support for (I've started a dialog with the correct people): dVRR and MRR.

- dVRR (discrete variable refresh rate) involves panels that have many discrete refresh rates, and can switch between them without a mode switch.  For example, there's a 120 Hz panel that can change the refresh duration in 1/240 (i.e. 4.168 msec) increments, from 8.33 msec up to some maximum.
- MRR (multi-refresh rate) involves a panel that has a few refresh rates, and can switch between them with a mode switch (e.g. 10 Hz, 60 Hz, 90 Hz, 120 Hz).

We'll definitely want dVRR addressed by this extension.  Again, I'm learning more and will get back with you.

Reply:
What's the diff? 4 ms increments is like 120, 80, 60, 48 Hz. That's basically the same few refresh rates. Feels like a distinction with very little difference.

Reply:
MRR sounds distinct from FRR, in that I don't see language in FRR that the refresh rate can change.

In Android, the operating system, can change display modes on the fly with MRR, without involvement from the application. For instance, a display normally running at 120Hz might instantly drop to 60Hz because of low battery conditions, or other contraints.

I think the point is that both dVRR and MRR are not quite VRR, because there are constraints not described in this extension, but they are not FRR either.